### PR TITLE
Solve animal tag relationship

### PR DESCRIPTION
**Pull request by damianooldoni**
This PR solves #88.
Function `get_animals()` updated, its documentation and its unit-tests.

Other small changes: function `as_tibble` imported from `dplyr` instead of `tibble` to reduce `etn` package's dependencies.

### CODE REVIEW

**Comment by damianooldoni** [comment]
I will now add a test about the collapsing of multiple values in `tag*` columns. Now there is only a test about uniqueness of animals (`pk`).

**Comment by damianooldoni** [comment]
Done.

**Comment by peterdesmet** [comment]
@damianooldoni can you explain why we need purrr, stringr and tidyselect?

**Comment by damianooldoni** [comment]
In order to test over all columns where collapsing can occur, i.e. all `tag*` columns.
See https://github.com/inbo/etn/blob/solve-animal-tag-relationship/tests/testthat/test-get_animals.R#L163-L172

I can use basic R `grepl` function instead of `str_detect` from `stringr` package to remove a dependency. And maybe grepl can also avoid using  rlang function `!!`? I can test it now.

**Comment by damianooldoni** [comment]
See https://github.com/inbo/etn/pull/106/commits/071764074a9d5ac44123bacf7137f8ffa7923022. I replaced functions of stringr, purr and rlang from `get_animals` and `test-get_animals`. 
The package `rlang` is however needed to load `.data`, which is in use in many pipes with dplyr functions (e.g. `df %>% filter(.data, condition)`). Using `.data` we don't get tens of notes while checking the package.

I think everything is now ok. Package remains very light, code is even more compact and functionality is added as requested.

File: R/zzz.R [hunk 1]
```diff
@@ -115,6 +115,8 @@ collapse_transformer <- function(regex = "[*]$", ...) {
 #'
 #' @keywords internal
 #'
+#' @noRd
```
Comment (peterdesmet) [inline R/zzz.R]: @damianooldoni Aha! Good to know.

**Pull request merged**